### ISSUE TITLE
SMB servers load 10x slower than in Windows' File Explorer

### ISSUE THREAD

**Issue by life036** [files-community/Files#12704]
### Description

Just like the title says. In addition, I have a folder on my local HD that just has shortcuts to about a dozen SMB servers (standard shortcuts to locations like \\ServerName), and it takes 30-40 seconds for that folder to load, vs instantly in Windows' File Explorer.

Something strange is going on with the way Files interacts with SMB servers.

### Steps To Reproduce

1. Connect to any SMB server on your LAN (\\ServerName) and compare how quickly it loads to the native file explorer.

2. Create a local folder on your HD with shortcuts to a handful of SMB servers on your network, and compare how quickly it loads to the native file explorer

### Requirements

SMB locations, and folders full of shortcuts to SMB locations, should load just as fast (or nearly as fast) as the native File Explorer.

### Files Version

2.5.1.0

### Windows Version

10.0.22621.1848

### Log File

[debug.log](https://github.com/files-community/Files/files/11852798/debug.log)

**Comment by Josh65-2201** [comment]
Thanks for the report. Can you give the following information.
- Is the issue with loading the contents of a network drive or the folder containing the links to the network drives
- How many items does the folder contain

**Comment by life036** [comment]
> Thanks for the report. Can you give the following information.
> 
> * Is the issue with loading the contents of a network drive or the folder containing the links to the network drives

Both situations have similar issues

> * How many items does the folder contain

My particular folder has 10 items, but the issue happens no matter how many shortcuts you have in there. It's just more pronounced when you have more items in the folder. It also seems to stutter longer on servers that aren't available; like it's checking the validity of the shortcut before loading it.

**Comment by Josh65-2201** [comment]
Thanks for the information, I've added this to the project board.

**Comment by KFlab** [comment]
Since I am a non-native English speaker, I am sorry for my poor English。
I have same problem and I think I found some useful information。

I found a lot of download traffic coming from my SMB server，
so I use wireshark to capture analysis.

Here's the information I got:

step 1) 
there's a <GetInfo Request> package try to read one of my file's info,
The target file is a mod pack for a game, and it's a 47.53GB size and zip type.
I will call this file "a.zip" 

step 2)
SMB server send a TCP package back 
and a <GetInfo Response> shows success

step 3)
There 's a <Read Request> to "a.zip" with Length 65536 Offset 65536

step 4)
There 's a <Read Request> to "a.zip" again with Length 69632 Offset 51039236096
The file here seems Central Directory Record part of a zip file

step 5) <<< i think here have some problem
Another <Read Request> to "a.zip" with Length 131072 Offset 131072

and more <Read Request> with Length 131072 or 262144

![Image](https://github.com/user-attachments/assets/ffab91e4-0456-4fd2-85a8-f209966a952e)

But prublem here is , the directory I'm opening is a directory of my Aria downloads.
So it has a lot of huge zipped files in it.

I haven't started reading more of the program's source code, and I'll try to spend a little bit of time locating the source of this traffic

**Comment by KFlab** [comment]
I have figured out what happened. When a directory starts with "\\", it is recognized as a shared directory. This type of directory has some special optimizations, so when browsing directories in this format, there will be no lag.

But when this directory is mapped as a network drive such as "Y:\"

The part of the program that handles compressed files will try to get the entire file
This causes IO blocking，and it crashes the program

Another problem comes from aria2
aria2 will create a temporary download file, such as "a.zip.aria2“
The part responsible for processing the compressed package will treat it as part of "a.zip"
This exacerbates the lag.

**Comment by SilverGreen93** [comment]
I think I have the same issue. If I open a network share folder containing around 160 files (exe and msi), the folder loads very slow, more than 10-20 seconds. In this time, the whole Files UI is not responding. In Windows Explorer or Directory Opus, the folder listing loads very quick.
This is more noticeable if the file listing is sorted by Date modified.

Files version: 3.9.12.0

Edit:
I think this happens because each exe files has to be read and load the icon to display in the list.
I tested with 600 zip files and it loads faster than these 160 executables.
In any case, Windows Explorer and other explorer apps load them faster, just a couple of seconds.

**Comment by c-michail** [comment]
@Josh65-2201 
instead of adding features you should focus in the loading speed of items to at least match windows explorer! Nobody gonna use this awesome app if it's slow!

Thank you

**Comment by Josh65-2201** [comment]
Stopping work on new features wouldn't improved the performance. Files is community developed, meaning people chose what they want to work and there is already work being done to make performance improvements happen, see #4180

**Comment by doctorrerumnaturalium** [comment]
Is there any news yet? I'm using Files 4.0.28.0. When connecting to a network drive, it still takes a very, very long time for folders and files to load and be displayed.  This issue has been open for more than two and a half years now...

**Comment by Josh65-2201** [comment]
When there is they will be posted here or https://github.com/files-community/Files/issues/4180

**Comment by SilverGreen93** [comment]
@c-michail and @doctorrerumnaturalium I feel your pain, I really abandoned using this app because of the slow performance compared to all other file managers in multiple places (network files, right-click menu etc), but I am keeping an eye on it and the progress made, cause when it will perform better, I might switch again back to it.

**Comment by doctorrerumnaturalium** [comment]
@SilverGreen93 I completely agree with you. It's a real shame, because I actually think the concept behind [Files](https://files.community/) is quite good. But unfortunately, it's completely unusable for everyday tasks...

**Comment by c-michail** [comment]
Seems the guys making this app don't care ....

**Comment by Josh65-2201** [comment]
@c-michail  We do. It isn't as simple as changing a toggle or easy one update fix. See https://github.com/files-community/Files/issues/4180 as it includes reasons on why and progress on updates related to performance

**Comment by c-michail** [comment]
@Josh65-2201  Well i don't code c# yo help you, but it seems it's design issue to me. Maybe you should start a new version with the correct design.

ps: If you can fix the performance to match at least windows explorer then it will the the best file manager!
